Add a blog post about state of affairs in OKD CI/CD (#31)
okd-project/okd.io: docs/blog/2023-07-18-State-of-Affairs-in-OKD-CI-CD.md

# State of affairs in OKD CI/CD

*by [Jakob Meng](https://github.com/jm1)*

date: 2023-07-18

> [OKD](https://www.okd.io/) is a community distribution of Kubernetes which is build from Red Hat OpenShift components on

Reviewer: ```suggestion
[OKD](https://www.okd.io/) is a community distribution of Kubernetes which is built from Red Hat OpenShift components on
```

top of Fedora CoreOS (FCOS) and recently also CentOS Stream CoreOS (SCOS). The OKD variant based on Fedora CoreOS is
called [OKD or OKD/FCOS](https://github.com/okd-project/okd). The SCOS variant is often referred to as [OKD/SCOS](
https://github.com/okd-project/okd-scos/).

The previous blog posts introduced [OKD Streams](2022-10-25-OKD-Streams-Building-the-Next-Generation-of-OKD-together.md)
and [its new Tekton pipelines](2022-12-12-Building-OKD-payload.md) for building OKD/FCOS and OKD/SCOS releases. This
blog post gives an overview of the current build and release processes for FCOS, SCOS and OKD. It outlines OKD's
dependency on OpenShift, an remnant from the past when its Origin predecessor was a downstream rebuild of OpenShift 3,
and concludes with an outlook on how OKD Streams will help users, developers and partners to experiment with future
OpenShift.

## Fedora CoreOS and CentOS Stream CoreOS

> [Fedora CoreOS is build with a Jenkins pipeline running in Fedora's infrastructure](

Reviewer: ```suggestion
[Fedora CoreOS is built with a Jenkins pipeline running in Fedora's infrastructure](
```

> https://github.com/coreos/fedora-coreos-pipeline) and is being maintained mostly by Red Hat's CoreOS team.

Reviewer 2: ```suggestion
https://github.com/coreos/fedora-coreos-pipeline) and is being maintained by the Fedora CoreOS team.
```

> CentOS Stream CoreOS is build with a [Tekton pipeline](https://github.com/okd-project/okd-coreos-pipeline/) running in a

Reviewer: ```suggestion
CentOS Stream CoreOS is built with a [Tekton pipeline](https://github.com/okd-project/okd-coreos-pipeline/) running in a
```

OpenShift cluster on [MOC](https://massopen.cloud/)'s infrastructure and pushed to `quay.io/okd/centos-stream-coreos-9`.

> Unlike FCOS and RHCOS, the SCOS build pipeline is owned and maintained by the OpenShift OKD Streams team and [SCOS

Reviewer 2: ```suggestion
The SCOS build pipeline is owned and maintained by the OpenShift OKD Streams team and [SCOS
```

are mirrored to `origin` namespaces on `quay.io`. [For example, at most every 3 hours a OCP 4.14 release image](
https://github.com/openshift/release/blob/master/core-services/release-controller/_releases/release-ocp-4.14-ci.json)
will be deployed (and upgraded) on AWS and GCP and afterwards tested with [OpenShift's conformance test suite](
https://github.com/openshift/origin). [When it passes the non-optional tests the release image and its dependencies will
be mirrored to `quay.io/origin` (except for `rhel-coreos*`, `*-installer` and some other images)](
https://github.com/openshift/release/blob/master/core-services/release-controller/_releases/release-ocp-4.14-ci.json).
These OCP CI releases are listed with a `ci` tag at [amd64.ocp.releases.ci.openshift.org](
https://amd64.ocp.releases.ci.openshift.org/). Builds and promotions of `nightly` and `stable` OCP releases are handled
differently (i.e. outside of Prow) by the [Automated Release Tooling (ART)](
https://source.redhat.com/groups/public/openshift/openshift_wiki/openshift_automated_release_tooling_art_team_faqs)

> team.

Reviewer: I have filed an RFE with the testplatform team to change this flow, and promote images directly to OKD without gating them on OCP CI releases: https://issues.redhat.com/browse/DPTP-3565

It might make sense to mention and link to that here.

Author: Ok, I have added a remark about release gating in braces. Better?

## OKD payload components

> A few payload components are build specifically for OKD though, for example OKD/FCOS' [okd-machine-os](

Reviewer: ```suggestion
A few payload components are built specifically for OKD though, for example OKD/FCOS' [okd-machine-os](
```

missed one :sweat_smile:

https://github.com/openshift/okd-machine-os). Unlike RHCOS and SCOS, [okd-machine-os](

> https://github.com/openshift/okd-machine-os), the operating system running at OKD/FCOS nodes,

Reviewer 2: ```suggestion
https://github.com/openshift/okd-machine-os), the operating system running on OKD/FCOS nodes,
```

[is layered on top of FCOS](https://github.com/openshift/okd-machine-os/blob/master/Dockerfile) (also
see [CoreOS Layering](https://github.com/coreos/enhancements/blob/main/os/coreos-layering.md),
[OpenShift Layered CoreOS](
https://github.com/openshift/enhancements/blob/master/enhancements/ocp-coreos-layering/ocp-coreos-layering.md)).

Note, some payload components have OKD specific configuration in OpenShift CI although the resulting images are not

> incorporated into OKD release images. For example, [OVN-Kubernetes images are being build and tested in OpenShift CI to

Reviewer: ```suggestion
incorporated into OKD release images. For example, [OVN-Kubernetes images are built and tested in OpenShift CI to
```
